crates/controller/src/mouse.rs

```diff
@@ -288,3 +319,6 @@ fn check_double_click(
         *last_click_position = Some(mouse_clicked.position());
     }
 }
+
+#[derive(Component)]
+pub(crate) struct HudVisibleNode;
```

Comment:
What about `HudTopVisibleNode`? Also, please document what this component means.

And I would place it alongside HUD as it is where it logically belongs.

Reply:
It does belong to HUD, but as you written somewhere before you try to have all systems used in one plugin in single file.. which makes me confused as to where to put this `HudTopVisibleNode` and the newly created `SystemParam` called `HudNodes`..

Reply:
And please where to document this? Is it sufficient in **code as a comment** eg. `\\\ Any HUD element that is the parent Node and should is not supposed to be draggable and other  interaction with in-game objects` or somewhere more detailed comment in documentation files like `CONTRIBUTING.md`?

Reply:
Code comment is sufficient. `Top-level non-transparent UI node. All such nodes are marked with this component and no descendants have it attached` or something like that.

Reply:
I would move this and the system parameter to the different module. The component does not have to be used from mouse whatsoever. It has to be temporarily public due to the limitations of system param derive, but that will change with Bevy v10.